### PR TITLE
[ORCA-223] / [ORCA-224] / [ORCA-225] Adds Submission Annotations

### PR DESCRIPTION
**Pull request by BWMac**
This PR adds the `ANNOTATE_SUBMISSION` module which is used to update the submission annotations after both the `VALIDATE` and `SCORE` steps. 

In order to facilitate the annotations, I updated the structure of `results.json` to only include top-level key-value pairs, so that each creates its own annotation. 

I also made a few changes to simplify the repository and overall workflow.

- Moved all modules remaining in `main.nf` to their own module scripts (including replacing the old `GET_SUBMISSIONS` module.
- Streamlined the `workflow` logic
- Created a more explicit DAG flow (represented in the diagram added to `README.md`

### CODE REVIEW

File: bin/score.py [hunk 1]
```diff
@@ -32,12 +32,9 @@ def score_submission(predictions_path: str, status: str) -> dict:
             score = None
             score_status = "INVALID"
     result = {
-        "score": {
-            "score_category": "auc",
-            "score": score,
-            "status": score_status,
-            "errors": message,
-        }
+        "auc": score,
```
Comment (thomasyu888) [inline bin/score.py:35]: Nit: In the future, i can imagine something like:

```
{"score_category": "auc", "score": "..." ...}
```

Edit... nevermind, sometimes there are multiple score categories...  Short of uploading semi-structured data... lets keep this.

File: main.nf [hunk 2]
```diff
@@ -11,88 +11,32 @@ params.cpus = "4"
...
-    UPDATE_SUBMISSION_STATUS_AFTER_RUN(RUN_DOCKER.output.map { tuple(it[0], "ACCEPTED") }, UPDATE_SUBMISSION_STATUS_BEFORE_RUN.output)
-    VALIDATE(RUN_DOCKER.output)
-    UPDATE_SUBMISSION_STATUS_AFTER_VALIDATE(VALIDATE.output.map { tuple(it[0], it[2]) }, UPDATE_SUBMISSION_STATUS_AFTER_RUN.output)
-    SCORE(VALIDATE.output)
-    UPDATE_SUBMISSION_STATUS_AFTER_SCORE(SCORE.output.map { tuple(it[0], it[2]) }, UPDATE_SUBMISSION_STATUS_AFTER_VALIDATE.output)
+    UPDATE_SUBMISSION_STATUS_BEFORE_RUN(image_ch.map { tuple(it[0], "EVALUATION_IN_PROGRESS") })
+    RUN_DOCKER(image_ch, SYNAPSE_STAGE.output, params.cpus, params.memory, UPDATE_SUBMISSION_STATUS_BEFORE_RUN.output)
+    UPDATE_SUBMISSION_STATUS_AFTER_RUN(RUN_DOCKER.output.map { tuple(it[0], "ACCEPTED") })
+    VALIDATE(RUN_DOCKER.output, UPDATE_SUBMISSION_STATUS_AFTER_RUN.output)
+    UPDATE_SUBMISSION_STATUS_AFTER_VALIDATE(VALIDATE.output.map { tuple(it[0], it[2]) })
```
Comment (thomasyu888) [inline main.nf:37]: Can you remind me what is in `it[1]`?

**Review by thomasyu888 — APPROVED** [review]
🍾 This is fantastic!  Just a note, the upcoming challenge we will set up a wireframe for will be:

1. Get submissions (this will be file-based)
2. Validate
3. update status / annotations
4. Score
5. update status / annotations

File: main.nf (hunk 2 above)
Comment (BWMac) [reply main.nf:37]: the path to the predictions file

**Pull request merged**